### PR TITLE
add-airbnb-eslint

### PR DESCRIPTION
**Pull request by mohit038**
#42  Adds eslint-Airbnb config
- [x] Prettier
- [x] Airbnb Eslint
- [x] recommended-requiring-type-checking
- [ ] review rules that has been turned off (left for follow-up PR)

### CODE REVIEW

**Review by tobiasdiez — CHANGES_REQUESTED** [review]
Nice! Did you already pair-programmed with @k3KAW8Pnf7mkmdSMPHz27 ?

One small suggestion: Please also use the plugin https://www.npmjs.com/package/eslint-plugin-unused-imports to detect unused imports. Otherwise LGTM.

File: package.json [hunk 1]
```diff
@@ -59,11 +59,16 @@
...
     "babel-loader": "^8.2.2",
     "clean-webpack-plugin": "^3.0.0",
     "copy-webpack-plugin": "^6.4.1",
     "css-loader": "^5.2.4",
     "eslint": "^7.29.0",
+    "eslint-config-airbnb": "^18.2.1",
+    "eslint-plugin-import": "^2.23.4",
+    "eslint-plugin-jsx-a11y": "^6.4.1",
     "eslint-plugin-office-addins": "^0.2.0",
     "eslint-plugin-react": "^7.24.0",
```
Comment (k3KAW8Pnf7mkmdSMPHz27) [inline package.json:73]: Oups, I think the instructions were missing the `react-hooks` dependency ([they suggest](https://www.npmjs.com/package/eslint-config-airbnb) a lower version number but 4.2 seems to work fine). Sorry about that, I have no idea why it didn't generate an error.

```suggestion
    "eslint-plugin-react": "^7.24.0",
    "eslint-plugin-react-hooks": "^4.2.0",
```

**Review by k3KAW8Pnf7mkmdSMPHz27 — CHANGES_REQUESTED** [review]
We might have to turn off more rules. @tobiasdiez, I just wanted to verify that this should be merged before fixing the ESLinter complaints in the files

**Comment by tobiasdiez** [comment]
Yes, I think it it's a good idea to merge this first and then fix the existing code later.

**Review by tobiasdiez — APPROVED** [review]

**Comment by k3KAW8Pnf7mkmdSMPHz27** [comment]
I think we need to turn the "regular" un-def off as well (see [the FAQ for motivation](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-get-errors-from-the-no-undef-rule-about-global-variables-not-being-defined-even-though-there-are-no-typescript-errors)). Either that or put the JSX global back on (sorry)

File: .eslintrc.json [hunk 2]
```diff
@@ -8,9 +8,49 @@
...
+    "eslint:recommended",
+    "plugin:@typescript-eslint/recommended",
+    "plugin:@typescript-eslint/recommended-requiring-type-checking",
+    "plugin:office-addins/react",
+    "airbnb",
+    "airbnb/hooks",
+    "prettier",
+    "plugin:prettier/recommended"
+    ],
+  "parser": "@typescript-eslint/parser",
```
Comment (k3KAW8Pnf7mkmdSMPHz27) [inline .eslintrc.json:25]: Perhaps also

```suggestion
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    }
  }
```

although it doesn't seem to have caused any issues having it off?

**Review by k3KAW8Pnf7mkmdSMPHz27 — APPROVED** [review]
Sorry about the delay. I think it looks good! If something more pop-ups we can address it in a new PR.

**Pull request merged**